### ISSUE TITLE
Incorrect translation of forecastTimeUnits for values other than "hours" on bounded forecast_period coordinates

### ISSUE THREAD

**Issue by s-boardman** [SciTools/iris-grib#321]
For data which should have a bounded `forecast_period` where the end of the period is greater than t+255 Grib changes the representation to have units of `3 hours` rather than `hours`. This is because the `P1` and `P2` values are stored as a single octet so would overflow if greater than 255. To work around this grib expects the reading software to multiply the `P1/2` values by the units to produce the _actual_ values. Therefore units of `3 hours` will produce the actual value of `P1=252` and `P2=258` for the relevant sections of the below grib header (output by `grib_dump`):

```
18        unitOfTimeRange = 10 [3 hours (grib1/4.table) ]
19        P1 = 84
20        P2 = 86
21        timeRangeIndicator = 2 [Product with a valid time ranging between reference time + P1 and reference time + P2 (grib1/local/ecmf/5.table , grib1/5.table) ]
```

When processing data from ECMWF that went beyond t+255 I noticed that cubes with bounded forecast periods were being produced with incorrect `forecast_period` coordinate, eg for the above grib record:

`forecast_period: 253.0 3 hours, bound=(252.0, 254.0) 3 hours`

It should be:

`forecast_period: 255.0 hours, bound=(252.0, 258.0) hours`

Debugging the `iris_grib._grib1_load_rules` module showed that if the generated `_forecastTimeUnit` of a grib message is anything other than `hours` then the `forecast_period` coordinate will be incorrect due to the the `add_bounded_time_coords` [function](https://github.com/SciTools/iris-grib/blob/main/iris_grib/_grib1_load_rules.py#L187) incorrectly calculating the period and setting the units.

When calculating the period to be used as the `forecast_period` the following code first converts the `t_bounds` (time_bounds) into hours and then tries to repeat the conversion when calculating the `period` from the `t_bounds` by using the original unit as shown below:

```python
t_bounds = grib.phenomenon_bounds('hours')
period = Unit('hours').convert(t_bounds[1] - t_bounds[0], grib._forecastTimeUnit)
```

When the `_forecastTimeUnit` is `hours` this conversion has a null effect. However, if this is 3 hours, for example, then the conversion is incorrect:

```python
>>> Unit('hours').convert(258 - 252, "3 hours")
2.0
```

Given that the conversion to hours has already taken place by the `phenomenon_bounds('hours') call, the period should be calculated as a simple subtraction of the lower bound from the upper:

```python
t_bounds = grib.phenomenon_bounds('hours')
period = t_bounds[1] - t_bounds[0]
```

In addition when the [units are set on the `forecast_period` coordinate](https://github.com/SciTools/iris-grib/blob/main/iris_grib/_grib1_load_rules.py#L193) uses the original `_forecastTimeUnit` string. This should be hard coded to `"hours"` as the conversion to `hours` is enforced by the calculations earlier in the function.

----------------

To test this specific case the following test code was added to the `iris_grib.tests.uni.grib1_load_rules.test_grib_convert.TestBoundedTime` test case:

```python
    def assert_bounded_message_3hours(self, **kwargs):
        attributes = {'productDefinitionTemplateNumber': 0,
                      'edition': 1, '_forecastTime': 252,
                      '_forecastTimeUnit': '3 hours',
                      'phenomenon_bounds': lambda u: (252, 258),
                      '_phenomenonDateTime': -1,
                      'table2Version': 9999,
                      '_originatingCentre': 'xxx',
                      }
        attributes.update(kwargs)
        message = mock.Mock(**attributes)
        self._test_for_coord(message, grib1_convert, self.is_forecast_period,
                             expected_points=[255],
                             expected_bounds=[[252, 258]])
        self._test_for_coord(message, grib1_convert, self.is_time,
                             expected_points=[255],
                             expected_bounds=[[252, 258]])
```

And each `test_time_range_indicator_N` method had the following added:

```python
def test_time_range_indicator_N(self):
        self.assert_bounded_message(timeRangeIndicator=N)
        self.assert_bounded_message_3hours(timeRangeIndicator=N)
```

---------------

For reference the corresponding lines of a `grib_dump` on a file with a `forecast_period` ending before 255 are as follows:

```
18        unitOfTimeRange = 1 [Hour (grib1/4.table) ]
19        P1 = 246
20        P2 = 252
21        timeRangeIndicator = 2 [Product with a valid time ranging between reference time + P1 and reference time + P2 (grib1/local/ecmf/5.table , grib1/5.table) ]
```

I have raw ECMWF data on the MO HPC to demonstrate on real world input as well. [Here's a reference to my internal investigations for those with access](https://github.com/MetOffice/stage/issues/418)

**Comment by s-boardman** [comment]
[Branch on my fork with a fix](https://github.com/s-boardman/iris-grib/tree/321-correct-bounded-forecast-period-calculation)

**Comment by lbdreyer** [comment]
Closed by #322
